### PR TITLE
Server address validation

### PR DESCRIPTION
**Pull request by melowe**
Added validator and validations for serverConfig.serverAddress and  serverConfig.bindingAddress.

### CODE REVIEW

File: config/src/main/java/com/quorum/tessera/config/constraints/ValidServerAddress.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/*
```
Comment (prd-fox) [inline config/src/main/java/com/quorum/tessera/config/constraints/ValidServerAddress.java]: Remove header

File: config/src/main/java/com/quorum/tessera/config/constraints/ValidServerAddress.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+@Retention(RUNTIME)
+@Constraint(validatedBy = ServerAddressValidator.class)
+@Documented
+public @interface ValidServerAddress {
+
+    boolean isBindingAddress() default false;
+    
+    String[] supportedSchemes() default {"unix","http","https"};
+    
+    String message() default "{ValidServerAddress.message}";
```
Comment (prd-fox) [inline config/src/main/java/com/quorum/tessera/config/constraints/ValidServerAddress.java]: Not sure if used, but either way this message doesn't exist.

File: config/src/main/java/com/quorum/tessera/config/constraints/ValidServerAddress.java (hunk 1 above)
Comment (prd-fox) [inline config/src/main/java/com/quorum/tessera/config/constraints/ValidServerAddress.java]: Remove header

File: config/src/main/java/com/quorum/tessera/config/constraints/ValidServerAddress.java (hunk 2 above)
Comment (prd-fox) [inline config/src/main/java/com/quorum/tessera/config/constraints/ValidServerAddress.java:26]: Not sure this is used, but this message doesn't exist. Is it worth defining in the validation message list?

File: config/src/main/java/com/quorum/tessera/config/constraints/ValidServerAddress.java (hunk 1 above)
Comment (melowe) [reply config/src/main/java/com/quorum/tessera/config/constraints/ValidServerAddress.java]: opps

**Review by prd-fox — APPROVED** [review]

**Pull request merged**